File: movie-backend/src/main/java/com/example/movie/dto/UpdateMovieRequest.java
package com.example.movie.dto;

import jakarta.validation.constraints.*;

public record UpdateMovieRequest(
        @NotEmpty @Size(min = 3, max = 30) String title,
        @NotEmpty @Size(min = 3, max = 100) String description,
        @NotNull @Min(10) @Max(300) Integer lengthInMinutes,
        @NotNull @Min(1850) @Max(2030) Integer releaseYear
) {
}

File: movie-frontend/src/app/movie/models/update-movie-request.ts
export class UpdateMovieRequest {
  constructor(
    public readonly title: string,
    public readonly description: string,
    public readonly lengthInMinutes: number,
    public readonly releaseYear: number,
  ) {}

  public static fromForm(
    form: UpdateMovieRequestRawFormValue,
  ): UpdateMovieRequest {
    return new UpdateMovieRequest(
      form.title,
      form.description,
      form.lengthInMinutes,
      form.releaseYear,
    );
  }
}

export interface UpdateMovieRequestRawFormValue {
  title: string;
  description: string;
  lengthInMinutes: number;
  releaseYear: number;
}
